docs2/site/docs/migrations/migration5.md

```diff
@@ -570,6 +570,36 @@ var options = new ExecutionOptions
 var result = await executer.ExecuteAsync(options);
 ```
 
+### 19. Subscription support improved
+
+Support for subscriptions has been moved from the `GraphQL.SystemReactive` nuget package directly into
+the main `GraphQL` package. There is no need to use `SubscriptionDocumentExecuter` (removed), and the default
+document executer will support subscriptions without overriding `SelectExecutionStrategy`.
+
+The new implementation of `SubscriptionExecutionStrategy` supports some new features and bug fixes:
+
+1. Serial execution of data events' field resolvers is supported by passing an instance of
+   `SerialExecutionStrategy` to the constructor. As before, parallel execution is default.
+
+2. Errors and output extensions are returned along with data events.
+
+3. Memory leaks have been eliminated in the case of errors, output extensions, metrics being enabled,
+   or the use of the context's array pool.
+
+4. The unhandled exception handler properly handles all error situations that it was designed to.
+
+5. The `System.Reactive` nuget reference is not necessary for GraphQL. You may still choose to use
+   `System.Reactive` nuget package in your library if you wish.
+
+6. Derived implementations allow for a scoped DI provider during execution of data events. It will
+   be necessary to override `ProcessDataAsync` and change the `RequestServices` property to a scoped
```

Comment:
Was now it scoped before?

Reply:
It has not changed, except as I will describe below.

The server project creates a scope for the execution of the initial request for an IObservable.  GraphQL.NET does not change or touch the RequestServices property -- neither before or after this PR.

Then for resolving during data events:

- Previously: RequestServices was set to null for any data events.

- Now: RequestServices is untouched, and for the server project would contain a reference to a disposed scope -- but that is due to the server project passing in a scope and disposing of it.  If the server project did not create a scope initially, it would have a valid reference to the root scope.  Which would be just fine for projects that do not use scoped services.

- New feature: Since the RequestServices property is untouched, it allows for a derived instance of SubscriptionExecutionStrategy to create a scope for data events and so provide a scoped service provider to the resolvers during data events.  This is what I would need for my implementations.

So this change is fully backwards compatible, since if you tried to use RequestServices previously, it would have been null.  Now it will be a disposed service provider (with the server project as it is now).

Reply:
OK. Two suggestions:
1. Put all these comments into code. It really has value.
2. A bit of backward incopatibility:
```
if (context.RequestServices == null)
{
    DoSome(new MyClass());
}
else
{
    DoSome(context.RequestServices.GetRequiredService<MyClass>()); // BOOM!
}
```

Reply:
True